model_server_tools/model_packaging/export_model_utils.py

```diff
@@ -0,0 +1,289 @@
+        with open(os.path.join(model_path, signature_file), 'r') as f:
+            data = json.loads(f.read())
+            data['inputs'][0]['data_name'] = input_data[0][0]
+        with open(os.path.join(model_path, signature_file), 'w') as f:
+            f.write(json.dumps(data, indent=2))
+
+        with open(os.path.join(model_path, symbol_file), 'w') as f:
+            f.write(sym.tojson())
+
+        save_dict = {('arg:%s' % k): v.as_in_context(mx.cpu()) for k, v in params.items()}
+        mx.nd.save(os.path.join(model_path, params_file), save_dict)
+        return symbol_file, params_file
+
+    @staticmethod
+    def generate_publisher(publisherargs):
+        publisher = Publisher(author=publisherargs.author, email=publisherargs.email)
+        return publisher
+
+    @staticmethod
+    def generate_engine(engineargs):
+        engine = Engine(engine_name=engineargs.engine)
+        return engine
+
+    @staticmethod
+    def generate_model(modelargs):
+        model = Model(model_name=modelargs.model_name, handler=modelargs.handler)
+        return model
+
+    @staticmethod
+    def generate_manifest_json(args):
+        """
+        Function to generate manifest as a json string from the inputs provided by the user in the command line
+        :param args:
+        :return:
+        """
+
+        publisher = ModelExportUtils.generate_publisher(args) if args.author and args.email else None
+
+        engine = ModelExportUtils.generate_engine(args)
+
+        model = ModelExportUtils.generate_model(args)
+
+        manifest = Manifest(runtime=args.runtime, engine=engine, model=model, publisher=publisher)
+
+        return str(manifest)
+
+    @staticmethod
+    def create_manifest_file(model_path, manifest):
+        """
+        Function creates a file called manifest.json under model_path/MAR-INF folder
+
+        :param model_path:
+        :param manifest:
+        :return:
+        """
+        mar_inf_path = os.path.join(model_path, MAR_INF)
+
+        if not os.path.exists(mar_inf_path):
+            os.makedirs(mar_inf_path)
+
+        manifest_path = os.path.join(mar_inf_path, MANIFEST_FILE_NAME)
+
+        with open(manifest_path, 'w') as m:
+            json.dump(manifest, m, indent=4)
+
+        return manifest_path
+
+    @staticmethod
+    def clean_temp_files(temp_files):
+        for f in temp_files:
+            os.remove(f)
+
+    @staticmethod
+    def zip(export_file, model_path, files_to_exclude):
+        with zipfile.ZipFile(export_file, 'w', zipfile.ZIP_DEFLATED) as z:
+            ModelExportUtils.zip_dir(model_path, z, set(files_to_exclude))
+
+    @staticmethod
+    def zip_dir(path, ziph, files_to_exclude):
+
+        """
+        This method zips the dir and filters out some files based on a expression
+        :param path:
+        :param ziph:
+        :param files_to_exclude:
+        :return:
+        """
+        unwanted_dirs = set(['__MACOSX', '__pycache__'])
```

Comment:
unwanted_dirs = {'__MACOSX', '__pycache__'}

Reply:
Done in a46fa5d65e8493d56fda684f0cca7f0e97d7cf5c